Option to change display name

Moyf opened the pull request:
I added another option, which support display property (like `title`) rather than the original note name:
![image](https://github.com/user-attachments/assets/8d515955-e2fc-4f74-8bc6-7b9814017c8e)

For example:
![image](https://github.com/user-attachments/assets/64c97902-4ed1-4bff-b23d-cdccdc19a8f0)

This `2-Area.md` note will display as its title:
![image](https://github.com/user-attachments/assets/828001fa-9c40-47a8-9132-0e24f9fdeb9d)

ahts4962 commented on src/navigationComponent.ts:
```diff
@@ -128,18 +182,6 @@ export class NavigationComponent extends Component {
 				})
 		);
-
-		linkStates.sort((a, b) => {
```
This code sorts the annotation links so that they appear in the order specified in the "Annotation strings" field in the settings. Removing this may change the current behavior.

Moyf replied:
Oh yes, you're right! 

Now I've added this sorting back, and after checking the whole commits, I think the Windsurf has removed too much things.
I'll try to re-organize the all codes tomorrow  : )

Now this PR includes:
- add a "DevMode" switch in plugin settings
- support middle mouse click to open link in new tab
- support customizable property value as "property links"
  - and compatible with wiki links with display text
- add a setting to remove duplicated link
- add a setting to display the `title`(or other property) of target links

ahts4962 reviewed (changes requested):
I think there is one place that needs to be corrected. Please see comments.

Moyf commented:
@ahts4962 Hello! Sorry for the delay, I've recovered some lost codes and added a new mapping setting:
![image](https://github.com/user-attachments/assets/35eba25f-1136-4834-b541-45e058c85d5b)

How it looks:
![image](https://github.com/user-attachments/assets/bbca2b44-9dc0-4d9c-9c1e-86843eea02d9)

Here is a conclusion of what I added, in case you need to write update logs:

1. The navigation header now could also display notes from properties (refer to `Property Mappings` in the plugin settings).
2. Added support for using the middle mouse button to open links in a new tab.
3. Users can now select a property (e.g., `title`) to serve as the display name instead of the note's file name.
4. An option has been added to remove duplicate links in the navigation header.

When you have time, please take a look at these changes to see if they align with your thoughts. If you have any feedback or suggestions, please let me know, and I will make further adjustments. 

If necessary, feel free to modify the code as you see fit!

Looking forward to those features being merged :)

Moyf commented:
> If necessary, feel free to modify the code as you see fit!
> 
> Looking forward to those features being merged :)

Forgot to mention, the style in the screenshots is changed by my CSS snippet, I didn't change the style in the code : )

ahts4962 commented:
@Moyf I have confirmed that it works fine. I will release it after some minor work.

By the way, is it ok to remove the developer mode option? I don't think it's better to reveal it to users in the release version.

Moyf commented:
> @Moyf I have confirmed that it works fine. I will release it after some minor work.
> 
> By the way, is it ok to remove the developer mode option? I don't think it's better to reveal it to users in the release version.

Yeah of course! Feel free to remove it  ; )

ahts4962 commented:
I have released these features in 1.1.0. Thanks for your contribution!

Moyf commented:
> I have released these features in 1.1.0. Thanks for your contribution!

Great news! Thank you!! 😄
